src/Contracts/ConfigContract.php

```diff
@@ -0,0 +1,14 @@
+<?php declare(strict_types=1);
+
+namespace StrictPhp\HttpClients\Contracts;
+
+/**
+ * keep constructor with default parameters
+ */
+interface ConfigContract
+{
+    /**
+     * @param static $object
+     */
+    public function initByDefault(ConfigContract $object): void;
```

Comment:
I think proper "english" word would be `initDefaults`, or `initFrom` is probably better

Reply:
I rename to `initFromDefaultConfig`